src/ray/object_manager/pull_manager.cc

```diff
@@ -568,6 +595,8 @@ void PullManager::UpdateRetryTimer(ObjectPullRequest &request,
 
   // Bound the retry time at 10 * 1024 seconds.
   request.num_retries = std::min(request.num_retries + 1, 10);
+  // Also reset the fetch expiration timer, since we just tried this pull.
+  request.expiration_time_seconds = 0;
```

Comment:
This was the change that I think should fix the bug. Basically, we need to reset the fetch expiration timer if we do a successful pull.

Reply:
ah this makes sense! Dask on Ray 1TB has a lot of inefficient pulling issues, so this was manifested here I guess...

Reply:
(it is like inefficient code path kind of helps us revealing issues lol)